### PR TITLE
Update help files to reflect macOS branding

### PR DESCRIPTION
**Pull request by brianlheim**
Apple has rebranded OS X as macOS. This PR updates the help files to reflect that.

### CODE REVIEW

**Comment by brianlheim** [comment]
Somewhat OT question: I'm not experienced enough with git to solve this myself—how do I bring my fork of SC up to date with supercollider/supercollider while avoiding that merge commit in my PRs? (bedd8b5 above)

**Comment by snappizz** [comment]
Assuming you have `origin` pointing to your fork and `upstream` pointing to this repo, do this *before* you branch:

```
git pull upstream master
git push origin master
git checkout master
git checkout -b topic/my-branch
```

If you already have a pesky merge commit, you can usually get rid of it by rebasing or cherrypicking.

**Comment by brianlheim** [comment]
Gotcha, thanks!

File: HelpSource/Classes/Document.schelp [hunk 1]
```diff
@@ -124,7 +124,7 @@ method:: standardizePath
 argument:: p
 The file system path to the directory. An instance of link::Classes/String::.
 discussion::
-If it is a relative path, expand it to an absolute path relative to your document directory. Expand tildes in path (your home directory), resolve symbolic links (but not aliases). Also converts from OS9 macintosh path format. See PathName for more complex needs.
+If it is a relative path, expand it to an absolute path relative to your document directory. Expand tildes in path (your home directory), resolve symbolic links (but not aliases). Also converts from macOS 9 path format. See PathName for more complex needs.
```
Comment (vivid-synth) [inline HelpSource/Classes/Document.schelp]: Maybe "macOS 9" is an anachronism?

File: HelpSource/Classes/Main.schelp [hunk 2]
```diff
@@ -51,7 +51,7 @@ Called after the class library has been compiled.
 discussion::
 This calls the superclass' startup, which among other things initializes the link::Classes/AppClock:: and the top-level link::Classes/Environment::.
 
-Main's startup then stores Server.default in the interpreter variable s, sets the platform default's link::Classes/GUI:: kit, calls a link::Classes/Platform:: specific startup method (for example OSXPlatform's startup opens the server windows), and finally invokes StartUp.run.
+Main's startup then stores Server.default in the interpreter variable s, sets the platform default's link::Classes/GUI:: kit, calls a link::Classes/Platform:: specific startup method (for example macOSPlatform's startup opens the server windows), and finally invokes StartUp.run.
```
Comment (vivid-synth) [inline HelpSource/Classes/Main.schelp]: Is this right?

File: HelpSource/Classes/Pipe.schelp [hunk 3]
```diff
@@ -37,7 +37,8 @@ returns:: The exit status of the command (an Integer).
 Examples::
 
 note::
-For anyone still using OS X 10.3, UNIX commands like pipe do not work when the server is booted; quit the server, otherwise SuperCollider crashes. More recent OS X is not affected.
+For anyone still using macOS 10.3, unix commands like pipe do not work when the server is booted; quit the server, otherwise sc crashes. More recent macOS is not affected.
```
Comment (vivid-synth) [inline HelpSource/Classes/Pipe.schelp]: looks like a regression on "unix" and "sc"

File: HelpSource/Classes/SCImage.schelp [hunk 4]
```diff
@@ -5,12 +5,12 @@ related:: Classes/SCImageFilter, Classes/SCImageKernel
 
 DESCRIPTION::
 
-SCImage is an image component for the Mac OS X SuperCollider client. SCImage is currently a wrapper around different models : you can use it for bitmap operations, image embedding for custom UI and for more advanced image processing as applying filters and kernels, both provided with the CoreImage framework.
+SCImage is an image component for the macOS supercollider client. SCImage is currently a wrapper around different models : you can use it for bitmap operations, image embedding for custom UI and for more advanced image processing as applying filters and kernels, both provided with the CoreImage framework.
```
Comment (vivid-synth) [inline HelpSource/Classes/SCImage.schelp]: lowercase "supercollider"

File: HelpSource/Classes/Main.schelp (hunk 2 above)
Comment (brianlheim) [inline HelpSource/Classes/Main.schelp]: Thought I had caught all the accidental changes of `OSXPlatform`, thanks for catching that one! Eventually the class should be renamed `MacOSPlatform`, but I wasn't intending to tackle that here.

File: HelpSource/Classes/Pipe.schelp (hunk 3 above)
Comment (brianlheim) [inline HelpSource/Classes/Pipe.schelp]: I think this branch needs to be rebased (?) to reflect the changes from #2592.

File: HelpSource/Classes/SCImage.schelp (hunk 4 above)
Comment (brianlheim) [inline HelpSource/Classes/SCImage.schelp]: same as above

File: HelpSource/Guides/News-3_5.schelp [hunk 5]
```diff
@@ -174,9 +174,7 @@ since no external FFT libraries are required.
 
 subsection:: Supernova
 
-A new multi-processor implementation of scsynth. Parallelism of the synthesis graph is exposed to the user via the
-link::Classes/ParGroup:: class. Supernova is currently Linux-only. It is not provided in the OSX/Windows binaries. In
-order to play patterns inside a ParGroup, the link::Classes/PparGroup:: can be used. Scsynth emulates parallel groups
+A new multi-processor implementation of scsynth. Parallelism of the synthesis graph is exposed to the user via the link::Classes/ParGroup:: class. Supernova is currently linux-only. It is not provided in the macOS/Windows binaries. In order to play patterns inside a ParGroup, the link::Classes/PparGroup:: can be used. Scsynth emulates parallel groups
```
Comment (vivid-synth) [inline HelpSource/Guides/News-3_5.schelp]: This should stay multiline

File: HelpSource/Guides/standalones.schelp [hunk 6]
```diff
@@ -3,11 +3,11 @@ SUMMARY:: A guide to creating SuperCollider-based applications
 CATEGORIES:: Platform>OSX, Tutorials
```
Comment (vivid-synth) [inline HelpSource/Guides/standalones.schelp]: looks like this category should change too?

File: HelpSource/Tutorials/Getting-Started/01-Introductory-Remarks.schelp [hunk 7]
```diff
@@ -5,7 +5,7 @@ related:: Tutorials/Getting-Started/00-Getting-Started-With-SC
 
 The following text is intended to serve as an introduction to SuperCollider 3, an object-oriented language for sound synthesis and digital signal processing (DSP). This tutorial does not assume a background in computer science, but does assume basic familiarity with your computer and its OS, as well as a basic knowledge of acoustics and digital audio. (I'm assuming here that words like frequency and sample will not cause any confusion.)
 
-The tutorial is written from a Mac OSX perspective, but much of it should apply to Linux and Windows as well. The parts which specifically differ have mostly to do with GUI aspects (Graphical User Interface).
+The tutorial is written from a macOS perspective, but much of it should apply to linux and windows as well. The parts which specifically differ have mostly to do with GUI aspects (Graphical User Interface).
```
Comment (vivid-synth) [inline HelpSource/Tutorials/Getting-Started/01-Introductory-Remarks.schelp]: "linux" and "windows" are lowercase again

File: HelpSource/Tutorials/JITLib/jitlib_networking.schelp [hunk 8]
```diff
@@ -15,8 +15,8 @@ subsection::before you start
 
 remember to synchronize your system clocks. This can be done by:
 definitionList::
-## in OS X || SystemPreferences>Date&Time: set emphasis::"Set Date & Time automatically":: to true.
-## in Linux || set the ntp clock
+## in macOS || SystemPreferences>Date&Time: set emphasis::"Set Date & Time automatically":: to true.
+## in linux || set the ntp clock
```
Comment (vivid-synth) [inline HelpSource/Tutorials/JITLib/jitlib_networking.schelp]: "linux"

File: HelpSource/Guides/standalones.schelp (hunk 6 above)
Comment (brianlheim) [inline HelpSource/Guides/standalones.schelp]: @vivid-synth would it be as simple as replacing all `Platform>OSX` with `Platform>macOS`?

File: HelpSource/Tutorials/Getting-Started/01-Introductory-Remarks.schelp (hunk 7 above)
Comment (brianlheim) [inline HelpSource/Tutorials/Getting-Started/01-Introductory-Remarks.schelp]: same as above re: my other PR

File: HelpSource/Tutorials/JITLib/jitlib_networking.schelp (hunk 8 above)
Comment (brianlheim) [inline HelpSource/Tutorials/JITLib/jitlib_networking.schelp]: same as above re: my other PR

File: HelpSource/Tutorials/Tutorial.schelp [hunk 9]
```diff
@@ -26,7 +26,7 @@ code::
...
+strong::VERY IMPORTANT:: : This line must be executed for the variable 'i' to be set. The mechanics are different depending on your platform. The macOS standard is to place the cursor anywhere on this line and press the "Enter" key on the numeric keypad. Pressing the main return key does not execute code! This allows you to write code fragments of multiple lines. To execute a multi-line block of code, select the block and press "Enter." For convenience, a code block can be enclosed in parentheses, and the entire block selected by double-clicking just inside either parenthesis. (For instructions in other editors (e.g. on Linux or Windows), consult the documentation specific to that platform. See also the helpfile link::Reference/KeyboardShortcuts:: for key commands in other editors.) If you don't have an enter key, then you can use ctrl-Return, Ctrl-c, fn-Return (on Some Macs), or Shift-Return.
```
Comment (vivid-synth) [inline HelpSource/Tutorials/Tutorial.schelp]: off-topic, but why is "Some" uppercase?

Comment (brianlheim) [inline HelpSource/Tutorials/Tutorial.schelp]: that's part of the rebrand too :P 
kidding, that should probably change and the shortcut names replaced with `Ctrl-Return, Ctrl-C, Fn-Return`

File: HelpSource/Guides/standalones.schelp (hunk 6 above)
Comment (vivid-synth) [inline HelpSource/Guides/standalones.schelp]: Not sure - anyone?

File: HelpSource/Classes/Pipe.schelp (hunk 3 above)
Comment (brianlheim) [inline HelpSource/Classes/Pipe.schelp]: Ok! I'm still learning the more advanced features of git so I have no idea what's easiest.

Comment (vivid-synth) [inline HelpSource/Classes/Pipe.schelp]: Alternately, if you fix all the ones I pointed out it's likely it'll merge cleanly even without a merge/rebase

File: HelpSource/Classes/Document.schelp (hunk 1 above)
Comment (vivid-synth) [inline HelpSource/Classes/Document.schelp]: I think this is referring to mac os 9 -- i.e. the version before 10.0

File: HelpSource/Guides/standalones.schelp (hunk 6 above)
Comment (brianlheim) [inline HelpSource/Guides/standalones.schelp]: I tried adding this category to a random .schelp file and it seems to have compiled no problem. I think Quarks tend to add their own categories as well so I see no issue here; would be great to have more input though.

screen shot 2016-12-27 at 1 51 54 pm

File: HelpSource/Classes/Pipe.schelp (hunk 3 above)
Comment (brianlheim) [inline HelpSource/Classes/Pipe.schelp]: Sure, I'll just do that.

**Comment by brianlheim** [comment]
@vivid-synth Fixed the diffs. Sorry for the separate commits but I don't currently have a SC repo on my machine and did it in-browser.

**Comment by telephon** [comment]
You can just write `MacOSPlatform : OSXPlatform {}`. But it will be a pretty useless entry in the method table, which we try to keep small. 

We need to make some tests some time, however, how much the method table is a performance issue.

**Comment by snappizz** [comment]
I don't think we should change OSX to macOS in any code yet. Not worth the trouble imo.

**Comment by brianlheim** [comment]
> I don't think we should change OSX to macOS in any code yet.

When will it be appropriate to do so, then?

**Comment by telephon** [comment]
The core classes of osx, aehm – sorry, macos, still are prefixed with `NS` for "NextStep"...  But we could start to deprecate it as soon as most people will assume this new naming as standard. Right now, nobody knows this, I fear, and would still look for osx, no?

**Comment by brianlheim** [comment]
10.12 Sierra has been out for a few months now, and 3.9 won't come out for a couple more (I think), so that should be plenty of time, especially if we deprecate.

**Comment by vivid-synth** [comment]
Maybe the solution here is to mark all searchable titles as both, e.g.

> macOS (Mac OS X)

(Or, if you prefer:)

> Mac OS X (macOS)

So that searching for either yields what you're looking for.

One argument for keeping this updated even though it doesn't particularly matter (e.g. @telephon 's point about NextStep) is that it can have a real effect on new users to have the sense that we're "in modern times," and not somehow behind the curve. This is a shiny new name, and if we use it we're shiny, if we don't we're last year's model, so to speak.

I don't use OS X (ha! I mean macOS) but I do think this matters.

**Comment by telephon** [comment]
the class is not used much, so it's fine to just deprecate it.

**Comment by snappizz** [comment]
This is ready to merge, right?

**Comment by brianlheim** [comment]
Changed the category name, if there are no concerns with that then this is ready to merge. Re: keeping up with the Joneses, that was one of my concerns actually.

**Comment by vivid-synth** [comment]
lgtm

**Pull request merged**